IotcRestApi/regulator_draft.md

```diff
@@ -0,0 +1,145 @@
+# API
+Bude se jednat o REST API, bude k dispozici Swagger.
+
+# Zabezpečení
+Komunikace bude zabezpečena bearer tokenem v hlavičce, informace budou doplněny.
+
+Většina endpointů přijímajících data v těle requestu vyžaduje položku `requestId`. Jedná se o jednoznačný identifikátor requestu, který generuje partner (jedná se o GUID) a slouží pro zajištění idempotence, čímž je docíleno toho, že akce je provedena jen jednou pokud dojde o opakovanému zavolání v určitém časovém intervalu.
+
+# Chybové odpovědi
+Chyby 5xx jsou způsobené chybou serveru a neměli by nikdy nastat. Pokud server vratí tuto chybu tak
+by měl být okamžitě informován zástupce z firmy Netlia.
+
+Chyby 4xx jsou vráceny, pokud klient provedl neplatný/nevalidní request. Vysvětlení stavových kódu je možné najít
+v HTTP dokumentaci - https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses. Podporované stavové kódy budou doplněny.
```

Comment:
Podporované stavové kódy budou doplněny. - to asi nebudou. Ja sam nevim jake vsechny muze ASP.NET vratit (bojim se ze to nebude zdokumentovane).

Navic bysme si uzavreli cestu kdybysme chteli pridat dalsi. Byla by to breaking zmena jelikoz partner by mohl rict ze tenhle konkretni stavovy kod jsme v dokumentaci neuvedli a on s nim nepocita.

Reply:
ok